Python/Product/TestAdapter.Executor/Pytest/TestDiscovererPytest.cs

```diff
@@ -91,12 +94,27 @@ internal class TestDiscovererPytest : IPythonTestDiscoverer {
             }
         }
 
-        private void CreateVsTests(IEnumerable<PytestDiscoveryResults> discoveryResults, ITestCaseDiscoverySink discoverySink) {
+        private void CreateVsTests(IEnumerable<PytestDiscoveryResults> discoveryResults,
+                                   ITestCaseDiscoverySink discoverySink,
+                                   Dictionary<string, PythonProjectSettings> sourceToProjectSettings) {
             foreach (var result in discoveryResults.MaybeEnumerate()) {
                 var parentMap = result.Parents.ToDictionary(p => p.Id, p => p);
                 foreach (PytestTest test in result.Tests) {
                     try {
-                        TestCase tc = test.ToVsTestCase(_settings.ProjectHome, parentMap);
+                        (string parsedSource, int line) = test.ParseSourceAndLine();
+                        var fullSourcePath = Path.IsPathRooted(parsedSource) ? parsedSource : Path.Combine(_settings.ProjectHome, parsedSource);
+                        if (!sourceToProjectSettings.ContainsKey(fullSourcePath)) {
+                            Warn(Strings.ErrorTestContainerNotFound.FormatUI(test.ToString()));
+                            continue;
+                        }
+
```

Comment:
sourceToProjectSettings is a dictionary that ignores case on key lookup so we can quickly tell if a source file doesn't have a matching testcontainer.  

However I dont have a nice way to get the TestContainers.Source, which is the non modified lowercase version.  So currently doing a linear search over the dictionary to get the matching key afterwards.